app/scripts/lib/channels/notifier.js

```diff
@@ -11,20 +11,56 @@ define(function (require, exports, module) {
 
   var _ = require('underscore');
   var Backbone = require('backbone');
+  var Validate = require('lib/validate');
 
   // Events that have the 'internal:' namespace should only be
   // handled by the content server. Other events may be handled
   // both externally and internally to the content server.
   var EVENTS = {
-    COMPLETE_RESET_PASSWORD_TAB_OPEN: 'fxaccounts:complete_reset_password_tab_open',
-    DELETE: 'fxaccounts:delete',
-    PROFILE_CHANGE: 'profile:change',
-    SIGNED_IN: 'internal:signed_in',
-    SIGNED_OUT: 'fxaccounts:logout'
+    COMPLETE_RESET_PASSWORD_TAB_OPEN: {
+      name: 'fxaccounts:complete_reset_password_tab_open',
+      schema: null
+    },
+    DELETE: {
+      name: 'fxaccounts:delete',
+      schema: { uid: 'String' }
+    },
+    PROFILE_CHANGE: {
+      name: 'profile:change',
+      schema: { uid: 'String' }
+    },
+    SIGNED_IN: {
+      name: 'internal:signed_in',
+      schema: {
+        // TODO: Eliminate sensitive fields from this data
+        email: '?String',
+        keyFetchToken: '?String',
+        lastLogin: '?Number',
+        needsOptedInToMarketingEmail: '?Boolean',
+        password: '?String',
```

Comment:
@shane-tomlinson, I thought #3390 would have eliminated the need for this but it hasn't yet. Were you expecting that, or do we need to investigate some more?